swap_meet/vendor.py

```diff
@@ -1,2 +1,80 @@
 class Vendor:
-    pass
\ No newline at end of file
+        def __init__(self, inventory = None):
+            if not inventory:
+                inventory = []
+            self.inventory = inventory
+
+        def add(self, item):
+            if not item in self.inventory:
+                self.inventory.append(item)
+                return item
+
+        def remove(self, item):
+            if item in self.inventory:
+                self.inventory.remove(item)
+                return item
+            else:
+                return False
+        
+        def get_by_category(self, category):
+            item_list = []
+            for item in self.inventory:
+                if category == item.category:
+                    item_list.append(item)
+            return item_list
+
+        def swap_items(self, vendor, my_item, their_item):
+            if my_item not in self.inventory or their_item not in vendor.inventory:
+                return False
+            self.inventory.remove(my_item)
+            vendor.remove(their_item)
+            self.inventory.append(their_item)
+            vendor.inventory.append(my_item)
+
+            return True
+        def swap_first_item(self, vendor):
+            if len(self.inventory) == 0 or len(vendor.inventory) == 0:
+                return False
+            # Get first item from inventory
+            first_item_inventory = self.inventory[0]
+            # Get first item from their inventory
+            first_item_vendor = vendor.inventory[0]
+            # Remove first item from inventory 
+            self.inventory.remove(first_item_inventory)
+            vendor.inventory.remove(first_item_vendor)
+            # "Update" first item on their inventory 
+            vendor.inventory.insert(0, first_item_inventory)
+            self.inventory.insert(0,first_item_vendor )
+            
+            
+            return True
+        
+        def get_best_by_category(self, category):
+            highest_item = None
+            higest_condition = 0
+            all_items = []
+            category = self.get_by_category(category)
+
+            if not category:
+                return None
+            
+
+            best_item = category[0]
```

Comment:
You can remove this check here on line 58 & 59. If there is nothing in `category`, then when you loop nothing will happen.

However, you set best_item to = category[0] on line 62.

So if category is empty, you would get an error trying to access the 0th element. So if you delete the check on lines 58 and 59, you'd need to set the variable `best_item = None` 

If you set `best_item = None` then if you loop and nothing happens because it's an empty list then you return None as per the requirements. If there is something in the list then `best_item` gets reassigned to the value of `item`